soh/src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -223,13 +347,21 @@ void func_809DF7D8(EnCow* this, GlobalContext* globalCtx) {
         this->actor.flags &= ~ACTOR_FLAG_16;
         Message_CloseTextbox(globalCtx);
         this->actionFunc = func_809DF778;
-        func_8002F434(&this->actor, globalCtx, GI_MILK, 10000.0f, 100.0f);
+        if (gSaveContext.n64ddFlag) {
+            GetItemID itemId = EnCow_GetRandomizerItemFromCow(this, globalCtx);
+            if (itemId != GI_ICE_TRAP) {
+                func_8002F434(&this->actor, globalCtx, itemId, 10000.0f, 100.0f);
+                EnCow_SetCowMilked(this, globalCtx);
+            }
```

Comment:
it seems like this will make it so the cow will always ice trap the player if the randomized item is an ice trap. it seems to me we should set the cow as milked even if we get ice trapped.

Reply:
So this is tricky, when it's an ice trap we don't want to attempt to give it to the player yet, we want the failsafe at 216 to instead give the ice trap due to weirdness with ice traps/flags. So with that being the case, if we set the cow as milked here, when the failsafe triggers to actually set off the trap `EnCow_GetRandomizerItemFromCow` will return `ITEM_NONE`, leading to no ice trap.

Sorry it's hard to explain, even harder to understand😆

Reply:
gotcha, so what happens if we just don't try giving anything in this function (considering the actionfunction is updated to be `func_809DF778`, why can't we just rely on that to do the item giving and ice trapping?)

Reply:
Great question, I think I tried this very early on and since I didn't have the ice trap issue sorted I wrote it off as not a possibility, but this works fine, updating PR